src/console/Startup.cs

```diff
@@ -106,8 +106,18 @@ public virtual StorageCheckpointClient ResolveCheckpointClient(IServiceProvider
             var checkpointBlobClient = factory.CreateStorageClient(checkpointContainerOptions);
             var logger = serviceProvider.GetRequiredService<ITelemetryLogger>();
 
-            storageOptions.BlobPrefix = applicationType;
-            var checkpointClient = new StorageCheckpointClient(checkpointBlobClient, storageOptions, logger);
+            var eventProcessorOptions = new EventHubClientOptions();
+            if (applicationType == _normalizationAppType)
+            {
+                Configuration.GetSection("InputEventHub").Bind(eventProcessorOptions);
+            }
+            else if (applicationType == _measurementToFhirAppType)
+            {
+                Configuration.GetSection("NormalizationEventHub").Bind(eventProcessorOptions);
+            }
+
+            storageOptions.BlobPrefix = $"{applicationType}/{storageOptions.BlobPrefix}";
```

Comment:
>BlobPrefix [](http://example.com/codeflow?start=76&length=10)

Is this getting set today?  Do we have examples?

Reply:
In the previous code this was only being set interally now we prefixing the application type in addition what is in the config.

Reply:
To Nate's comment if we allow this here the root can change and clean up could be prevented.

Reply:
So there is a property `"Checkpoint:BlobPrefix"` that was intended to be used as the blob prefix and which is parsed into storageOptions.BlobPrefix, but was actually not being honored, given that we were overriding it with app type. 
Today we seem to be hardcoding this value in PaaS to "devicedata"
https://microsofthealth.visualstudio.com/Health/_git/health-paas?path=%2Fsrc%2FResourceProviderApplication%2FResourceProvider.Iot.Workspace%2FModels%2FNormalizationEnvironmentVariables.cs&version=GBmaster&line=32&lineEnd=32&lineStartColumn=13&lineEndColumn=49&lineStyle=plain&_a=contents

It seemed like we wanted to expose this as an input but we weren't using it, so decided to append it to the app type (since we also want to scope the checkpoints to the app type as well). 
As explained in the other comment, with this change, we will not be cleaning up checkpoints up until this point that were in *app type*/checkpoint, but the data ingestion will be reset since its looking at the new path. Which I believe is okay? (given that we decided offline its potentially okay given we don't have many "production" Gen2 IoT connectors setup)
As for the property being taken as input, given that its hardcoded, I don't see the value personally, but maybe it was intended probably for people using the OSS ingest console app to be able to configure this property?

Reply:
The reason we do not honor Checkpoint:BlobPrefix is because we deploy OSS on Webjobs. Unfortunately because both of the Normalize and MeasurementToFhir Webjobs run on a single App Service, this means the environment variables are shared. Because the environment variables are shared, we run into issues where we need Normalization and MeasurementToFhir environment variables like Checkpoint:BlobPrefix and Console:ApplicationType to be set to different values. We could have created new environment variables for each application, or solved this another way, but happened to choose to override the two application specific variables Checkpoint:BlobPrefix and Console:ApplicationType variables at startup.

The main purpose for Checkpoint:BlobPrefix was just so we didn't hardcode anything in the class and had a way to to pass the prefix in as a parameter, but the intent was still to set upstream depending on the application type.

Reply:
Agreed, that makes sense for it to be different per app type. Which is why I retained that and just appended the passed in blob prefix param value.
As discussed offline with Will, we can technically do without blob prefix being passed in as an env variable, since we care about differentiating it based on app type. But say if in the future, we have more "data" types per app type, then appending that config driven blob prefix would help.
So the path should look like - Normalization/devicedata/*eventHubFQDN*/*eventHubName*

Reply:
Thanks I really appreciate the discussion here.  This all makes sense :)  Can we have a comment (or comments) where appropriate so the expected path is clear.  Part of my concern if we have different layers touching the path and result takes a bit to decipher.